### PR TITLE
Change get top of leaderboard to be by ranking points and added top winners

### PR DESCRIPTION
**Pull request by bkzhang**

### CODE REVIEW

File: middleware/crud-functions/crud-function-tests/postWinnersTest.js [hunk 1]
```diff
@@ -20,12 +20,17 @@ describe("postWinners", function() {
...
     revertWinData(["nash"]);
+
+    const revertedUser = await getUser("nash");
+
+    assert.equal(revertedUser["wins"], beforeUser["wins"]);
+    assert.equal(revertedUser["winStreak"], beforeUser["winStreak"]);
   });
 });
 
 function revertWinData(userHandles) {
```
Comment (bkzhang) [inline middleware/crud-functions/crud-function-tests/postWinnersTest.js]: @nashanator17 `admin.firestore` isn't a thing anymore

File: middleware/crud-functions/crud-function-tests/postLosersTest.js [hunk 2]
```diff
@@ -8,6 +8,22 @@ describe('postLosers', function() {
...
+    revertWinData(["nash"]);
+
+    const revertedUser = await getUser("nash");
+
+    assert.equal(revertedUser["losses"], beforeUser["losses"]);
+    assert.equal(revertedUser["lossStreak"], beforeUser["lossStreak"]);
+  });
 });
 
 function revertLossData(userHandles) {
```
Comment (bkzhang) [inline middleware/crud-functions/crud-function-tests/postLosersTest.js]: @nashanator17 `admin.firestore` pls

File: middleware/crud-functions/crud-function-tests/getTopOfLeaderboardTest.js [hunk 3]
```diff
@@ -14,6 +14,8 @@ describe("getTopOfLeaderboard", function() {
     user["losses"].should.be.a("number");
     user["winStreak"].should.be.a("number");
     user["lossStreak"].should.be.a("number");
+    user["rankScore"].should.be.a("number");
+    user["rankScore"].should.be.a("number");
```
Comment (joeyzhoucode) [inline middleware/crud-functions/crud-function-tests/getTopOfLeaderboardTest.js]: why is there two of these

File: middleware/crud-functions/crud-function-tests/getTopWinnersTest.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+const getTopWinners = require("../getTopWinners.js");
+const assert = require("chai").assert;
+const should = require("chai").should();
+
+describe("getTopWinners", function() {
+  it("should return a user object with valid attributes", async function() {
+    users = await getTopWinners(1);
```
Comment (joeyzhoucode) [inline middleware/crud-functions/crud-function-tests/getTopWinnersTest.js]: I think we should redeclare users with `let users` it's safer to scope it to each test.

File: middleware/crud-functions/crud-function-tests/postLosersTest.js [hunk 5]
```diff
@@ -8,6 +8,22 @@ describe('postLosers', function() {
...
+
+  it("should increment losses, winStreak and set lossStreak to 0", async function() {
+    const beforeUser = await getUser("nash");
+    await postWinners(["nash"]);
+    const afterUser = await getUser("nash");
+
+    assert.equal(afterUser["losses"], beforeUser["losses"]+1);
+    assert.equal(afterUser["lossStreak"], beforeUser["lossStreak"]+1);
+    assert.equal(afterUser["lossStreak"], 0);
+    revertWinData(["nash"]);
```
Comment (joeyzhoucode) [inline middleware/crud-functions/crud-function-tests/postLosersTest.js]: Use `afterEach()` so we don't have to retype `revert` for every test.

**Review by joeyzhoucode — APPROVED** [review]
LGTM.

File: middleware/crud-functions/crud-function-tests/postLosersTest.js (hunk 5 above)
Comment (bkzhang) [reply middleware/crud-functions/crud-function-tests/postLosersTest.js]: we pass in a value which might change so imo its better to keep it explicit so we don't assume that the data will revert if we have different winners in a new test

**Pull request merged**